osu.Game/Overlays/Rankings/SpotlightSelector.cs

```diff
@@ -0,0 +1,165 @@
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.API.Requests.Responses;
+using osuTK;
+using System;
+
+namespace osu.Game.Overlays.Rankings
+{
+    public class SpotlightSelector : Container
+    {
+        private readonly Box background;
+        private readonly SpotlightsDropdown dropdown;
+        private readonly DimmedLoadingLayer loading;
+
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        public readonly Bindable<APISpotlight> SelectedSpotlight = new Bindable<APISpotlight>();
+
+        private readonly InfoColumn startDateColumn;
+        private readonly InfoColumn endDateColumn;
+
+        public SpotlightSelector()
+        {
+            RelativeSizeAxes = Axes.X;
+            Height = 100;
+            Children = new Drawable[]
+            {
+                background = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Padding = new MarginPadding { Horizontal = UserProfileOverlay.CONTENT_X_MARGIN, Vertical = 10 },
+                    Children = new Drawable[]
+                    {
+                        dropdown = new SpotlightsDropdown
+                        {
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                            RelativeSizeAxes = Axes.X,
+                            Current = SelectedSpotlight,
+                            Depth = -float.MaxValue
+                        },
+                        new FillFlowContainer
+                        {
+                            Anchor = Anchor.BottomRight,
+                            Origin = Anchor.BottomRight,
+                            AutoSizeAxes = Axes.Both,
+                            Direction = FillDirection.Horizontal,
+                            Spacing = new Vector2(15, 0),
+                            Children = new Drawable[]
+                            {
+                                startDateColumn = new InfoColumn(@"Start Date"),
+                                endDateColumn = new InfoColumn(@"End Date"),
+                            }
+                        }
+                    }
+                },
+                loading = new DimmedLoadingLayer(),
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            background.Colour = colours.GreySeafoam;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            SelectedSpotlight.BindValueChanged(onSelectionChanged);
+        }
+
+        public void FetchSpotlights()
```

Comment:
The selector should not be doing the web request. It should:
1. Take in a set of spotlights.
2. Handle selecting a spotlight.
3. Return a spotlight.

The web request would be part of the overlay, and in this case the test scene providing the spotlights (the test scene should provide a custom set of spotlights).

Reply:
Got it